scripts/travis/es-integration-test.sh

```diff
@@ -1,39 +1,86 @@
 #!/bin/bash
 
-set -e
+set -euxf -o pipefail
 
-GOARCH=$(go env GOARCH)
+usage() {
+  echo $"Usage: $0 <es_version>"
+  exit 1
+}
 
-run_integration_test() {
-  ES_VERSION=$1
-  docker pull docker.elastic.co/elasticsearch/elasticsearch:${ES_VERSION}
-  CID=$(docker run --rm -d -p 9200:9200 -e "http.host=0.0.0.0" -e "transport.host=127.0.0.1" -e "xpack.security.enabled=false" -e "xpack.monitoring.enabled=false" docker.elastic.co/elasticsearch/elasticsearch:${ES_VERSION})
-  if [ "$ES_OTEL_INTEGRATION_TEST" == true ]; then
-    make es-otel-exporter-integration-test
-  else
-    STORAGE=elasticsearch make storage-integration-test
-    make index-cleaner-integration-test
+check_arg() {
+  if [ ! $# -eq 1 ]; then
+    echo "ERROR: need exactly one argument"
+    usage
   fi
-  docker kill $CID
 }
 
-run_integration_test "5.6.16"
-run_integration_test "6.8.2"
-run_integration_test "7.3.0"
+setup_es() {
+  local tag=$1
+  local image=docker.elastic.co/elasticsearch/elasticsearch
+  local params=(
+    --rm
+    --detach
+    --publish 9200:9200
+    --env "http.host=0.0.0.0"
+    --env "transport.host=127.0.0.1"
+    --env "xpack.security.enabled=false"
+    --env "xpack.monitoring.enabled=false"
+  )
+  local cid=$(docker run ${params[@]} ${image}:${tag})
+  echo ${cid}
+}
+
+setup_query() {
+  local arch=$(go env GOARCH)
+  local params=(
+    --es.tls=false
+    --es.version=7
+    --es.server-urls=http://127.0.0.1:9200
+    --query.bearer-token-propagation=true
+  )
+  SPAN_STORAGE_TYPE=elasticsearch ./cmd/query/query-linux-${arch} ${params[@]}
+}
+
+teardown_es() {
+  local cid=$1
+  docker kill ${cid}
+}
+
+teardown_query() {
+  local pid=$1
+  kill -9 ${pid}
+}
+
+build_query() {
+  make build-crossdock-ui-placeholder
+  GOOS=linux make build-query
+}
+
+run_integration_test() {
+  local es_version=$1
+  local cid=$(setup_es ${es_version})
+  STORAGE=elasticsearch make storage-integration-test
+  make index-cleaner-integration-test
+  make es-otel-exporter-integration-test
+  teardown_es ${cid}
```

Comment:
Doesn't token propagation test need a running ES? you seem to be stopping it prior to that.

Maybe ES setup/teardown should be in main?

Reply:
based on the logs the ES container is definitely being killed before the token test, so how does it pass? does it not need storage?

Reply:
Looking at the older version of the script we kill the ES container before token propagation test. I haven't looked into the details of token propagation test to see what it does though. I will check that to make sure.

Reply:
It looks like token propagation test mocks ES calls and requires only jaeger-query to be available.